frontend/src/components/Dashboard/page.tsx

```diff
@@ -14,45 +13,28 @@ const Dashboard = () => {
   const [selectedLocationPop, setSelectedLocationPop] = useState(null);
   const [expandedLocations, setExpandedLocations] = useState({});
   const [showPopup, setShowPopup] = useState(false);
-  const [memos, setMemos] = useState({});
-  const [reloadDashboard, setReloadDashboard] = useState(true);
+  const [memos, setMemos] = useState<any[]>([]);
```

Comment:
we know the type of memos being fetched, why is this `any`?

Reply:
originally I use MemoType but it doesn't save the memo._id
handleDelete 
setMemos(prevMemos => prevMemos.filter(m => m._id !== memo._id));
won't be able to use m._id to compare with the delete memo_id

Reply:
I'm not sure what you mean by this. 

`MemoType`s `id` field is optionally null, if that was the error you were facing, then you can apply `!` after it to force the TS compiler that it isn't null. 

Otherwise, when you are at the stage to delete a memo, the memo id should exist.

Reply:
so when the memo was created as MemoType, it has no id since id is generated after the post request.
That's why here when I fetch the memo as MemoType, they have no memo.id, so instead of making it MemoType I made it any so that i can access memo._id

Reply:
I sorted it out, const [memos, setMemos] = useState<MemoType[]>([]);